src/common/spawner.ts

```diff
@@ -0,0 +1,50 @@
+import { Event, EventEmitter } from 'vscode';
+
+export interface Spawnee {
+	stdout: Event<string>;
+	stderr: Event<string>;
+	stdin(data: string): void;
+	exit: Event<number>;
+	killed: boolean;
+	kill(): void;
+}
+
+
+export interface Spawner {
+	/**
+	 * Spawn a python `process`
+	 *
+	 * @param args: arguments to pass to the python
+	 * @returns an object that can be listened to for stdout/stderr/stdin
+	 */
+	spawn(args: string[], cwd: string | undefined): Promise<Spawnee>;
```

Comment:
I think? you can pass args to a worker? If not a message could be created to pass the args to pdb inside the worker.